.github/workflows/ci.yml

```
@@ -80,44 +80,32 @@ jobs:
           # Make sure the help options print.
           python -m pyinstaller -h
 
-      - name: Run tests
+      - name: Start display server
         if: startsWith(matrix.os, 'ubuntu')
         run: |
           Xvfb :99 &
-          export DISPLAY=:99
-          export QT_DEBUG_PLUGINS=1
-
-          pytest -n 3 --maxfail 3 --durations 10 tests/unit tests/functional \
-            --force-flaky --no-flaky-report --junitxml=junit-results.xml || true
-
-          # The ``run_tests`` script is invoked with the ``-c`` option to
-          # specify a ``pytest.ini``, rather than allowing pytest to find
-          # something unexpected in the filesystem (it searches from the root
-          # dir all the way to the top of the filesystem per
-          # https://docs.pytest.org/en/latest/customize.html).
-          python -m PyInstaller.utils.run_tests -c PyInstaller/utils/pytest.ini --include_only=pyi_hooksample.
+          echo "DISPLAY=:99" >> $GITHUB_ENV
+          echo "QT_DEBUG_PLUGINS=1" >> $GITHUB_ENV
 
       - name: Run tests
-        if: startsWith(matrix.os, 'macos')
-        run: |
-          pytest -n 3 --maxfail 3 --durations 10 tests/unit tests/functional --ignore tests/functional/test_libraries.py \
-            --force-flaky --no-flaky-report --junitxml=junit-results.xml || true
+        if: matrix.os != 'macos-latest'
```

Can we keep these as soft `startsWith()` conditions?
```suggestion
        if: ${{ !startsWith(matrix.os, 'macos') }}
```
The reason being that occasionally we like to try specific OS versions like `macos-11.0` which go wrong here. Same with the next one down.